beetmoverscript/tests/test_integration.py

```diff
@@ -0,0 +1,228 @@
+                            ],
+                            "checksums_path": "",
+                        },
+                    },
+                    "locale": "en-US",
+                    "taskId": "XGsGR0qdRmaWNE9fW5CmKQ",
+                },
+            ],
+            "releaseProperties": {"appName": "nightly_components"},
+            "upstreamArtifacts": [
+                {
+                    "paths": [
+                        "public/build/browser-awesomebar-63.0.20201013153553-sources.jar.asc",
+                        "public/build/browser-awesomebar-63.0.20201013153553.aar.asc",
+                        "public/build/browser-awesomebar-63.0.20201013153553.pom.asc",
+                    ],
+                    "taskId": "ZYfA8OtXRZuDeF5Xq3ubxg",
+                    "taskType": "signing",
+                },
+                {
+                    "paths": [
+                        "public/build/browser-awesomebar-63.0.20201013153553-sources.jar",
+                        "public/build/browser-awesomebar-63.0.20201013153553-sources.jar.md5",
+                        "public/build/browser-awesomebar-63.0.20201013153553-sources.jar.sha1",
+                        "public/build/browser-awesomebar-63.0.20201013153553.aar",
+                        "public/build/browser-awesomebar-63.0.20201013153553.aar.md5",
+                        "public/build/browser-awesomebar-63.0.20201013153553.aar.sha1",
+                        "public/build/browser-awesomebar-63.0.20201013153553.pom",
+                        "public/build/browser-awesomebar-63.0.20201013153553.pom.md5",
+                        "public/build/browser-awesomebar-63.0.20201013153553.pom.sha1",
+                    ],
+                    "taskId": "XGsGR0qdRmaWNE9fW5CmKQ",
+                    "taskType": "build",
+                },
+            ],
+        },
+    }
+
+
+def create_dummy_artifacts_for_task(task, root_path):
+    """Create artifacts to upload as specified in the given task."""
+    for artifact_dict in task["payload"]["artifactMap"]:
+        task_id = artifact_dict["taskId"]
+        for path in artifact_dict["paths"]:
+            full_path = root_path / task_id / path
+            full_path.parent.mkdir(parents=True, exist_ok=True)
+            create_dummy_artifact(full_path)
+
+
+def create_dummy_artifact(file_path):
+    with open(file_path, "wb") as f:
+        f.write(b"some data")
+
+
+@pytest.fixture
+def boto3_client_mock(monkeypatch):
+    client_mock = Mock(spec=beetmoverscript.script.boto3.client("s3"))
+    client_mock.generate_presigned_url.return_value = "presigned_url"
+
+    def fake_boto3_client_only_s3(*args, **kwargs):
+        assert args[0] == "s3", "A client other than 's3' was requested"
+        return client_mock
+
+    monkeypatch.setattr(beetmoverscript.script.boto3, "client", fake_boto3_client_only_s3)
+
+    return client_mock
```

Comment:
TIL `@pytest.fixture` work with return statements too! I've always used `yield`.

Reply:
I learned the opposite when I read your code :smile:! Turns out that `yield` can be used to [allow for teardown code in a fixture](https://docs.pytest.org/en/stable/fixture.html#fixture-finalization-executing-teardown-code), so using `return` or `yield` is equivalent when it's the last statement of the fixture.